jax/experimental/compilation_cache/gfile_cache.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pathlib
+
+from jax.experimental.compilation_cache.cache_interface import CacheInterface
+import tensorflow.io.gfile as gfile
+from absl import logging
+
+class GFileCache(CacheInterface):
+  
+  def __init__(self, path: str):
+    """Sets up a cache at 'path'. Cached values may already be present."""
+    self._path = pathlib.Path(path)
+    gfile.makedirs(self._path)
+  
+  def get(self, key: str):
+    """Returns None if 'key' isn't present."""
+    if not key:
+      raise ValueError("key cannot be empty")
+    path_to_key = str(self._path / key)
+    if gfile.exists(path_to_key):
+      with gfile.GFile(path_to_key, "rb") as file:
+        return file.read()
+    else:
+      return None
+
+  def put(self, key: str, value: bytes):
+    """Adds new cache entry."""
+    if not key:
+      raise ValueError("key cannot be empty")
+    path_to_new_file = str(self._path / key)
+    logging.info(path_to_new_file)
+    with gfile.GFile(path_to_new_file, "w") as f:
+      f.write(value)
+
+  def initialize_gfilecache(path, max_cache_size_bytes=None):
+    """Creates GFileCache Instance."""
+    assert cc._cache is None, f"The cache path has already been initialized to {cc._cache._path}"
+    if max_cache_size_bytes is not None:
```

Comment:
Let's just remove the max_cache_size_bytes argument. This will also require updating the tests.